src/components/button/button.tsx

```diff
@@ -0,0 +1,36 @@
+import { MouseEventHandler } from "react";
```

Comment:
I'd prefer if this was one export for `Button` and allowed a `variant` prop to be passed in to select which type of button is being used. We also need to account for inherent button attributes, such as `type`. Ideally this should be able to accept any attribute that you can put on a `<button>` element. We can do all of this like so:

```
import cx from "classnames";
import { HTMLAttributes, MouseEventHandler } from "react";

type Props = HTMLAttributes<HTMLButtonElement> & {
  variant?: "primary" | "secondary" | "text" | "icon";
  buttonText?: string;
  onClick?: MouseEventHandler;
  icon?: string;
  className?: string;
};

export const Button = ({
  variant = "primary",
  buttonText = "value",
  icon,
  className,
  ...rest
}: Props) => {
  return (
    <button
      {...rest}
      className={cx(
        `...`, // These are the base styles that apply to all button types
        variant === "primary" && `...`, // Styles for Primary button
        variant === "secondary" && `...`, // Styles for Secondary button
        variant === "text" && `...`, // Styles for Text button
        variant === "icon" && `...`, // Styles for Icon button
        className, // Additional classes passed from the parent component
      )}
    >
        {icon ? (
            <img className="p-1" src={icon}/>
        ) : ( buttonText )}
    </button>
  );
};
```